nomad/structs/structs.go

```diff
@@ -11074,6 +11074,21 @@ func (a *Allocation) NextRescheduleTimeByTime(t time.Time) (time.Time, bool) {
 	return a.nextRescheduleTime(t, reschedulePolicy)
 }
 
+// ShouldBeReplaced tests an alloc for replace in case of disconnection
+func (a *Allocation) ShouldBeReplaced() *bool {
+	tg := a.Job.LookupTaskGroup(a.TaskGroup)
+
+	if tg == nil {
+		return nil
+	}
+
+	if tg.Disconnect == nil || tg.Disconnect.Replace == nil {
```

Comment:
We're returning `nil` on line 11089 anyways in this case, so we can skip a comparison:

```suggestion
	if tg.Disconnect == nil {
```